src/Result.php

```diff
@@ -16,16 +18,63 @@ public function isValid(): bool
         return $this->errors === [];
     }
 
-    public function addError(string $message): void
+    /**
+     * @psalm-param list<int|string>|null $valuePath
+     */
+    public function addError(string $message, ?array $valuePath = null): void
```

Comment:
Test with collision:

$data = [
  'names' => [
    'god89',
    'greatJonh',
    'vasya77',
  ],
];
$rules = [
  'names' => [
    Each::rule(new Rules([
      MatchRegularExpression::rule('~\d~')->not()->skipOnError(false),
    ])),
    Number::rule()->skipOnError(false),
  ],
];

$nestedResult = Nested::rule($rules)->validate($data);
$this->assertSame([
  'names' => [
    ???
  ],  
], $nestedResult->getNestedDetailedErrors());

What result do we expect?

`names` contain both attributes with errors and errors themselves.

Ideas::

1) Remove errors of `name` itself:

0 => [
  'Value is invalid. god89 given.',
],
2 => [
  'Value is invalid. vasya77 given.',
],

2) Add errors in `names` (bad variant, can be collision with keys):

0 => [
  'Value is invalid. god89 given.',
],
2 => [
  'Value is invalid. vasya77 given.',
],
3 => 'Value must be a number.',

3) Throw exception.

Reply:
After discussion in messenger we decided to postpone tree implementation for the next iteration.

Reply:
:heavy_check_mark: